app/models/async_transaction/vet360/base.rb

```diff
@@ -2,6 +2,70 @@
 
 module AsyncTransaction
   module Vet360
-    class Base < AsyncTransaction::Base; end
+    class Base < AsyncTransaction::Base
+
+      FINAL_STATUSES = [
+        'REJECTED',
+        'COMPLETED_SUCCESS',
+        'COMPLETED_NO_CHANGES_DETECTED',
+        'COMPLETED_FAILURE',
+        #'RECEIVED_DEAD_LETTER_QUEUE' #@TODO Do we know what this is?
+      ]
+
+      # Updates the status and transaction_status with fresh API data
+      # @params user [User] the user whose tx data is being updated
+      # @params service [Vet360::ContactInformation::Service] an initialized vet360 client
+      # @params tx_id [int] the transaction_id
+      # @returns [AsyncTransaction::Vet360::Base]
+      def self.refresh_transaction_status(user, service, tx_id = nil)
+
+        transaction_record = Base.find_by(user_uuid: user.uuid, transaction_id: tx_id)
+
+        raise Common::Exceptions::RecordNotFound, transaction_record unless transaction_record
```

Comment:
Is there a `find_by` alternative we can use that will raise RecordNotFound for us automatically?

Reply:
I think adding a `!` will raise a `ActiveRecord::RecordNotFound`